### DISCUSSION TITLE
[NBug] fatal: bad object refs/heads/feature/package-manager

### DISCUSSION THREAD

**Discussion by tom-englert** [gitextensions/gitextensions#11128] [Q&A]
### Environment

- Git Extensions 4.1.0.16698
- Build 1fe52a137acef1fbd2b653dea263b44ad5844177
- Git 2.40.1.windows.1
- Microsoft Windows NT 10.0.19045.0
- .NET 6.0.18
- DPI 96dpi (no scaling)
- Portable: False

### Issue description

Exit code: 128
Command: C:\Program Files\Git\bin\git.exe
Arguments: -c log.showSignature=false log -z --pretty=format:"%H%T%P%n%at%n%ct%n%aN%n%aE%n%cN%n%cE%n%B" --max-count=100000 --exclude=refs/notes/commits --all --
Working directory: D:\Develop\My Open Source\GitHub\NuGetMonitor\

```
GitExtUtils.ExternalOperationException: fatal: bad object refs/heads/feature/package-manager
 ---> System.Exception: fatal: bad object refs/heads/feature/package-manager
   --- End of inner exception stack trace ---
   at void GitUI.RevisionGridControl.PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs, bool forceRefresh)+() => { } in C:/projects/gitextensions/GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:line 1309
   at async Task GitUI.GitUIExtensions.InvokeAsync(Control control, Action action, CancellationToken token) in C:/projects/gitextensions/GitUI/GitUIExtensions.cs:line 239
   at async void GitUI.TaskManager.FileAndForget(Task task, Func<Exception, bool> fileOnlyIf)+(?) => { } in C:/projects/gitextensions/GitExtUtils/GitUI/TaskManager.cs:line 29
```

just opened a repo:
[git.zip](https://github.com/gitextensions/gitextensions/files/12267533/git.zip)

### Steps to reproduce

The repository that causes the crash is attached.

### Did this work in previous version of GitExtensions?

 It worked fine when I was working on it a few days ago. I'm not aware of any updates since this.

**Comment by RussKie** [comment]
You have [.git/FETCH_HEAD file](https://stackoverflow.com/q/9237348/2338036), which is suggesting you were running `git pull` operation which got interrupted and git failed to reset that file. This got your repository in a broken state.

```bash
PS> git --no-optional-locks for-each-ref --sort="-*authordate" --sort="-authordate" --format="%(if)%(authordate)%(then)%(objectname) %(refname)%(else)%(*objectname) %(*refname)%(end)"

warning: ignoring broken ref refs/heads/feature/package-manager
ebc4a7aa7ee34c278a1fd77a0afb1b675196a677 refs/remotes/origin/feature/package-manager
18255fd05ea9759ff7988bc9cd406b7c70526a93 refs/heads/temp2
...

PS> git -c log.showSignature=false log -z --pretty=format:"%H%T%P%n%at%n%ct%n%aN%n%aE%n%cN%n%cE%n%B" --max-count=100000 --exclude=refs/notes/commits --all --

fatal: bad object refs/heads/feature/package-manager
```

This is something you need to fix yourself. https://stackoverflow.com/q/11834108/2338036 or something similar may help.

**Reply by tom-englert** [reply]
@RussKie I somewhat expected that the repo was broken, I already fixed this - however I thought that maybe GitExtensions should not crash, but just show the error message and let me open another repo.

**Reply by gerhardol** [reply]
Before GE 4.0 these errors were silently ignored, which could result in other issues.
GE is not crashing, it does not know how to proceed here. Maybe the error message for "fatal; bad object" should not offer to report an issue, it is not relevant to report.

Some error messages like this could be suppressed or handled, but this is a pure Git error.
